rbx_types/src/tags.rs

```diff
@@ -0,0 +1,108 @@
+use std::convert::TryFrom;
+use std::str::FromStr;
+use std::string::FromUtf8Error;
+
+#[derive(Default, Clone, Debug, Eq, PartialEq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(transparent)
+)]
+pub struct Tags {
+    tag_list: Vec<String>,
+}
+
+impl Tags {
+    pub fn new() -> Tags {
+        Self {
+            tag_list: Vec::new(),
+        }
+    }
+
+    pub fn as_vec(&self) -> &Vec<String> {
+        &self.tag_list
+    }
+
+    pub fn as_mut_vec(&mut self) -> &mut Vec<String> {
+        &mut self.tag_list
+    }
+}
+
+impl From<Vec<String>> for Tags {
+    fn from(tag_list: Vec<String>) -> Tags {
+        Self { tag_list }
+    }
+}
+
+impl From<Vec<&str>> for Tags {
+    fn from(tag_list: Vec<&str>) -> Tags {
+        Self {
+            tag_list: tag_list
+                .iter()
+                .map(|tag_name| String::from_str(tag_name).unwrap())
+                .collect(),
+        }
+    }
+}
+
+impl From<&Tags> for Vec<u8> {
+    fn from(tags: &Tags) -> Vec<u8> {
+        let tag_list = &tags.tag_list;
+        let mut buf: Vec<u8> = Vec::with_capacity(tag_list.capacity() + tag_list.len());
+
+        for tag_name in tag_list.as_slice() {
+            buf.extend_from_slice(tag_name.as_bytes());
+            buf.extend_from_slice(b"\0")
+        }
+
+        buf
+    }
+}
+
+impl TryFrom<Vec<u8>> for Tags {
+    type Error = FromUtf8Error;
+
+    fn try_from(buf: Vec<u8>) -> Result<Tags, FromUtf8Error> {
+        let len = buf.len();
+        let mut tag_list = Vec::new();
+        let mut count = 0;
+
+        while count < len {
+            let tag_name: Vec<u8> = buf
+                .iter()
+                .skip(count)
+                .copied()
+                .take_while(|element| {
+                    count += 1;
+                    *element != 0
+                })
+                .collect();
+
+            tag_list.push(String::from_utf8(tag_name)?);
+        }
+
+        Ok(Self { tag_list })
+    }
+}
```

Comment:
You can use [`slice::split`](https://doc.rust-lang.org/std/primitive.slice.html#method.split) and save a lot of effort here.

Similar to the `From` implementation, this should probably be its own method instead of a generic trait implementation.